PhysicsTools/TensorFlow/interface/TBBThreadPool.h

```diff
@@ -27,7 +27,7 @@ namespace tensorflow {
     }
 
     explicit TBBThreadPool(int nThreads = -1)
-        : nThreads_(nThreads > 0 ? nThreads : tbb::task_scheduler_init::default_num_threads()), numScheduleCalled_(0) {
+        : nThreads_(nThreads > 0 ? nThreads : tbb::global_control::max_allowed_parallelism), numScheduleCalled_(0) {
```

Comment:
I suspect this is not correct. The `tbb::global_control::max_allowed_parallelism` is an enumeration
```cpp
class global_control {
public:
    enum parameter {
        max_allowed_parallelism,
```
https://github.com/oneapi-src/oneTBB/blob/9e15720bc7744f85dff611d34d65e9099e077da4/include/oneapi/tbb/global_control.h#L57-L60
and the code here should use the actual number of threads (that is communicated for TensorFlow via the `NumThreads()` function).

On the other hand, I suspect the `default_num_threads()` to be potentially incorrect as well (I'd naively expect the `NumThreads()` needing to return the number of threads that can be used, not the number of logical cores on the machine).

Reply:
Here is some past discussion
https://github.com/cms-sw/cmssw/pull/28711#issuecomment-579233555
https://github.com/cms-sw/cmssw/pull/28711#issuecomment-579283411
https://github.com/cms-sw/cmssw/pull/28711#issuecomment-579288424

This PR should not get too involved in "what is correct here" though.

Reply:
How about `tbb::global_control::active_value(tbb::global_control::max_allowed_parallelism)`
https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/FWCore/Services/plugins/InitRootHandlers.cc#L854

Reply:
@dan131riley pointed to
https://github.com/cms-sw/cmssw/blob/b4bb14f6573e803ecbe516d306d651d9774d48b7/FWCore/Services/plugins/InitRootHandlers.cc#L852-L854
I think we should do the same here (and leave any further tuning for later).